chainer/functions/loss/discriminative_loss.py

```diff
@@ -0,0 +1,437 @@
+        prediction, labels, n_objects, gt_idx,
+        delta_v, delta_d, max_n_clusters,
+        norm=1, alpha=1.0, beta=1.0, gamma=0.001):
+    """Discriminative margin-based clustering loss function
+
+    This is the implementation of the following paper:
+    https://arxiv.org/abs/1802.05591
+
+    In segmentation, one of the biggest problem is to have noise at the output
+    of a trained network.
+    For cross-entropy based approaches, if the pixel value is wrong,
+    the loss value will be same independent from the wrong pixel's location.
+    Even though the network gives wrong pixel output, it is desirable
+    to have it as close as possible to the original position.
+    By applying discriminative loss function, groups of segmentation instances
+    can be moved together.
+    This loss function calculates the following three parameters:
+
+    - Variance Loss:
+        Loss to penalize distances between pixels which are belonging
+        to same instance. (Pull force)
+    - Distance loss:
+        Loss to penalize distances between the centers of instances.
+        (Push force)
+    - Regularization loss:
+        Small regularization loss to penalize weights against overfitting.
+
+    Args:
+        prediction(:class:`~chainer.Variable` or \
+                   :class:`numpy.ndarray` or \
+                   :class:`cupy.ndarray`) :
+                   segmentation prediction output
+                   (batch size, total instance count, width, height)
+        labels(:class:`~chainer.Variable` or \
+               :class:`numpy.ndarray` or \
+               :class:`cupy.ndarray`) :
+               segmentation ground truth
+               (batch size, total instance count, width, height)
+        n_objects(:class:`~chainer.Variable` or \
+                  :class:`numpy.ndarray` or \
+                  :class:`cupy.ndarray`) :
+                  number of objects in ground truth
+                  (batch size, )
+        gt_idx(:class:`~chainer.Variable` or \
+               :class:`numpy.ndarray` or \
+               :class:`cupy.ndarray`) :
+               indexes of non-zero ground truths
+               (batch size, variable length)
+        delta_v (float): Minimum distance to start penalizing variance
+        delta_d (float): Maximum distance to stop penalizing distance
+        max_n_clusters (int): Maximum possible number of clusters.
+        norm (int): Norm to calculate pixels and cluster center distances
+        alpha (float): Weight for variance loss      (alpha * variance_loss)
+        beta (float): Weight for distance loss       (beta * distance_loss)
+        gamma (float): Weight for regularization loss(gamma * regularizer_loss)
+
+    Returns:
+        (:class:`~chainer.Variable` or \
+        :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            (alpha * variance_loss) +
+            (beta * distance_loss) +
+            (gamma * regularizer_loss)
+    """
```

Comment:
Please update this docstring to be identical to the one used for `DiscriminativeMarginBasedClusteringLoss`, since they should be the same.

Reply:
I am quite confused with this one. The wrapper contains initialization + result parameter which is obtained by calling __call__ function of `DiscriminativeMarginBasedClusteringLoss` class. However, class documentation only contains the initialization information. Hence, I am planning to remove output value from `DiscriminativeMarginBasedClusteringLoss` class. What do you think?

Reply:
Oh, sorry, this is my mistake. You're right that the docstrings should not be identical. What I meant was that both docstrings contain some identical paragraphs, so I meant that the typos that were fixed in the other docstring should also be fixed in this one.